packages/react-components/react-aria/library/src/useTypingAnnounce/useTypingAnnounce.ts

```diff
@@ -0,0 +1,65 @@
+import * as React from 'react';
+import { useTimeout } from '@fluentui/react-utilities';
+import { useAnnounce, useFluent_unstable as useFluent } from '@fluentui/react-shared-contexts';
+import type { AnnounceOptions } from '@fluentui/react-shared-contexts';
+import { AriaLiveAnnounceFn } from '../AriaLiveAnnouncer/AriaLiveAnnouncer.types';
+
+type Message = {
+  message: string;
+  options: AnnounceOptions;
+};
+
+const valueMutationOptions = {
+  attributes: true,
+  subtree: true,
+  characterData: true,
+  attributeFilter: ['value'],
+};
+
+export const useTypingAnnounce = (inputEl: React.RefObject<HTMLElement>): { typingAnnounce: AriaLiveAnnounceFn } => {
+  const { targetDocument } = useFluent();
+  const { announce } = useAnnounce();
+
+  const observer = React.useRef<MutationObserver>();
+  const [setTypingTimeout, clearTypingTimeout] = useTimeout();
+  const messageQueue = React.useRef<Message[]>([]);
+
+  const callback: MutationCallback = React.useCallback(
+    (mutationList, mutationObserver) => {
+      clearTypingTimeout();
+      setTypingTimeout(() => {
+        messageQueue.current.forEach(({ message, options }) => {
+          announce(message, options);
+        });
+        messageQueue.current.length = 0;
+        mutationObserver.disconnect();
+      }, 500);
+    },
+    [announce, clearTypingTimeout, setTypingTimeout],
+  );
+
+  const typingAnnounce: AriaLiveAnnounceFn = React.useCallback(
+    (message: string, options: AnnounceOptions = {}) => {
+      const win = targetDocument?.defaultView;
+      if (!win || !inputEl?.current) {
+        return;
+      }
+
+      messageQueue.current.push({ message, options });
+
+      if (!observer.current) {
+        // create mutation observer if it doesn't exist
+        observer.current = new win.MutationObserver(callback);
+      }
```

Comment:
Let's move creation of `MutationObserver` to a separate `useEffect()` i.e.

```
React.useEffect(() => {
    const targetWindow = targetDocument?.defaultView;
    if (!targetWindow) {
      return;
    }

  const observer= targetWindow win.MutationObserver(callback);

 observer.current = observer;

  return () => {
        observer.disconnect();
   }
}, [callback])
```

This way we ensure that we there will be issues in concurrent mode